features/FxA-45-device-registration-api/README.md

```diff
@@ -0,0 +1,371 @@
+# Device registration API
+
+https://mozilla.aha.io/features/FXA-45
+
+* [Stories](#stories)
+  * [As an API consumer](#as-an-api-consumer)
+  * [As a user](#as-a-user)
+* [Data](#data)
+* [API](#api)
+  * [POST /v1/account/create](#post-v1accountcreate)
+  * [POST /v1/account/login](#post-v1accountlogin)
+  * [POST /v1/account/device](#post-v1accountdevice)
+  * [GET /v1/account/devices](#get-v1accountdevices)
+  * [POST /v1/account/device/destroy](#post-v1accountdevicedestroy)
+* [Details](#details)
+  * [How are unrecognised devices handled?](#how-are-unrecognised-devices-handled)
+  * [What is a disconnected device?](#what-is-a-disconnected-device)
+
+## Stories
+
+### As an API consumer
+
+* As an API consumer,
+  I want to create an account with the auth server
+  using new device details
+  and receive the newly-registered device id in the response.
+
+* As an API consumer,
+  I want to log in to an auth server account
+  using new device details
+  and receive the newly-registered device id in the response.
+
+* As an API consumer,
+  I want to log in to an auth server account
+  using a registered device id.
+
+* As a logged-in API consumer,
+  I want to register new device details
+  and receive the newly-registered device id in the response.
+
+* As a logged-in API consumer,
+  I want to update the details for a registered device.
+
+* As a logged-in API consumer,
+  I want to get a list of all the registered device details.
+
+* As a logged-in API consumer,
+  I want to deregister a registered device
+  and remove all of its details from the server.
+
+* As a logged-in API consumer,
+  I want to receive a `"new sync device"` push notification
+  when a new device is registered.
+
+* As a logged-in API consumer,
+  I want to receive a `"device name changed"` push notification
+  when a registered device is renamed.
+
+* As a logged-in API consumer,
+  I want to receive a `"password changed"` push notification
+  when the password is changed.
+
+* As a logged-in API consumer,
+  I want to receive a `"password reset"` push notification
+  when the password is reset.
+
+* As a logged-in API consumer,
+  I want to receive a `"disconnected"` push notification
+  when a registered device transitions to the disconnected state.
+
+### As a user
```

Comment:
One tiny nit we be that the user stories should come first, but at that point we're well off into the process-over-product weeds :-)

I think this document completely fulls its purported purpose of "first draft" and we should merge it, and continue to iterate on it concurrently with the implementation.  Thanks for pushing this through @philbooth!